src/displayapp/screens/Twos.cpp

```diff
@@ -97,39 +92,37 @@ Twos::~Twos() {
 }
 
 bool Twos::placeNewTile() {
-  std::vector<std::pair<int, int>> availableCells;
-  for (int row = 0; row < 4; row++) {
-    for (int col = 0; col < 4; col++) {
-      if (!grid[row][col].value) {
-        availableCells.push_back(std::make_pair(row, col));
-      }
+  unsigned int emptyCells[nCells];
```

Comment:
using `std::array` feels more like modern C++ than C-raw-arrays. Any reason why it was removed? It should be basically syntactic sugar

Reply:
It was using a vector and more complicated than necessary.

Reply:
See JF's reply above.

Reply:
for 2D data I understand JF's reply. But for 1D data like it is here I would say `std::array<unsigned int, nCells> emptyCells;` is just as readable

```suggestion
  std::array<unsigned int, nCells> emptyCells;
```

Reply:
It doesn't really make a difference, but since std::array doesn't have any benefits in this case, I'm not sure it's worth changing.